HomeComponents/Buttons/UpdateButton.py

```diff
@@ -13,15 +13,16 @@
 import sys
 from urllib import request    
 from pkg_resources import parse_version 
+import threading
+
 class UpdateButton(Button):
 	def __init__(self, parent):
 		super(UpdateButton, self).__init__(parent)
 		self.osr2mp4_current_ver = pkg_resources.get_distribution("osr2mp4").version
 		self.osr2mp4app_current_ver = pkg_resources.get_distribution("osr2mp4app").version
-
-		pool = ThreadPool(processes=1)
-		async_result = pool.apply_async(self.check_updates)
-
+		x = threading.Thread(target=self.check_updates)
+		x.start()
+		x.join()
```

Comment:
literally useless thread if you put join here. join() will wait the thread to finish before executing the next codes. So it's basically not thread at all.